From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 23 Jun 2023 15:40:00 +0200
Subject: [PATCH] Remove some unnecessary dependencies

- org.junit.* are testing dependencies; these exclusions don't remove them all,
  I don't know how are they getting to the final build.
- quarkus-vertx-http-dev-ui-resources are some Quarkus dev tools that pulls in
  74 org.mvnpm.* dependencies with JS libraries.
  See https://github.com/keycloak/keycloak/issues/22399.
- quarkus-jdbc-mssql and quarkus-jdbc-oracle are for proprietary databases,
  we don't support those.

There are still many useless dependencies, but it's not easy to get rid of them.

--- a/quarkus/deployment/pom.xml
+++ b/quarkus/deployment/pom.xml
@@ -21,6 +21,17 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-core-deployment</artifactId>
+            <!-- XXX-Patched: added -->
+            <exclusions>
+                <exclusion>
+                    <groupId>org.junit.platform</groupId>
+                    <artifactId>junit-platform-launcher</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.junit.jupiter</groupId>
+                    <artifactId>junit-jupiter</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
@@ -40,6 +40,13 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-reactive-routes-deployment</artifactId>
+            <!-- XXX-Patched: added -->
+            <exclusions>
+                <exclusion>
+                    <groupId>io.quarkus</groupId>
+                    <artifactId>quarkus-vertx-http-dev-ui-resources</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
@@ -103,6 +103,7 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <!-- XXX-Patched
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jdbc-mssql-deployment</artifactId>
@@ -123,6 +124,7 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        -->
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-bootstrap-core</artifactId>
--- a/quarkus/runtime/pom.xml
+++ b/quarkus/runtime/pom.xml
@@ -59,6 +59,7 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jdbc-mysql</artifactId>
         </dependency>
+        <!-- XXX-Patched
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jdbc-mssql</artifactId>
@@ -67,6 +68,7 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jdbc-oracle</artifactId>
         </dependency>
+        -->
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-core</artifactId>
